src/Triple/Extra.elm

```diff
@@ -0,0 +1,256 @@
+values contained in a Triple.
+
+    ( 1, 2, Array.fromList [0,1,2,3,4] )
+        |> Triple.Extra.apply Array.slice
+        --> Array.fromList [1,2,3]
+
+-}
+apply : (a -> b -> c -> d) -> ( a, b, c ) -> d
+apply f ( a, b, c ) =
+    f a b c
+
+
+{-| Similar to List.sortWith but for Triples instead of lists. Instead of
+converting values contained in a Triple to `comparable`s, instead supply a
+function that will produce an `Order` directly.
+
+    Triple.Extra.sortWith Basics.compare ( 2, 1, 3 )
+        --> ( 1, 2, 3 )
+
+-}
+sortWith : (a -> a -> Order) -> ( a, a, a ) -> ( a, a, a )
+sortWith toOrder ( a, b, c ) =
+    case ( toOrder a b, toOrder b c ) of
+        ( LT, LT ) ->
+            ( a, b, c )
+
+        ( GT, GT ) ->
+            ( c, b, a )
+
+        ( EQ, EQ ) ->
+            ( a, b, c )
+
+        ( LT, EQ ) ->
+            ( a, b, c )
+
+        ( EQ, LT ) ->
+            ( a, b, c )
+
+        ( EQ, GT ) ->
+            ( c, a, b )
+
+        ( GT, EQ ) ->
+            ( c, b, a )
+
+        ( GT, LT ) ->
+            case toOrder a c of
+                GT ->
+                    ( b, c, a )
+
+                _ ->
+                    ( b, a, c )
+
+        ( LT, GT ) ->
+            case toOrder a c of
+                GT ->
+                    ( c, a, b )
+
+                _ ->
+                    ( a, c, b )
+
+
+
+-- Mapping ---------------------------------------------------------------------
+
+
+{-| Transform the first value in a triple.
+
+    Triple.Extra.mapFirst negate ( -3, 10, '9' )
+        --> ( 3, 10, '9' )
+
+-}
+mapFirst : (a -> x) -> ( a, b, c ) -> ( x, b, c )
+mapFirst f ( a, b, c ) =
+    ( f a, b, c )
+
+
+{-| Transform the second value in a triple.
+
+    Triple.Extra.mapSecond negate ( -3, 10, '9' )
+        --> ( -3, 010, '9' )
+
+-}
+mapSecond : (b -> x) -> ( a, b, c ) -> ( a, x, c )
+mapSecond f ( a, b, c ) =
+    ( a, f b, c )
+
+
+{-| Transform the third value in a triple.
+
+    Triple.Extra.mapThird String.toUpper ( -3, 10, "a" )
+        --> ( -3, 10, "A" )
+
+-}
+mapThird : (c -> x) -> ( a, b, c ) -> ( a, b, x )
+mapThird f ( a, b, c ) =
+    ( a, b, f c )
+
+
+{-| Apply a function to all values contained in a Triple.
+
+    Triple.Extra.map negate ( -3, 10, -7 )
+        --> ( 3, -10, 7 )
+
+-}
+map : (a -> b) -> ( a, a, a ) -> ( b, b, b )
+map f ( a, b, c ) =
+    ( f a, f b, f c )
+
+
+
+-- List conversion --------------------------------------------------------------
+
+
+{-| Turns a triple into a list of three elements.
+
+    Triple.Extra.toList ( 1, 2, 3 )
+        --> [ 1, 2, 3 ]
+
+-}
+toList : ( a, a, a ) -> List a
+toList ( a, b, c ) =
+    [ a, b, c ]
+
+
+{-| Creates a triple from the first three elements of a list.
+
+    Triple.Extra.fromList [ 1, 2, 3, 4 ]
+        --> Just ( 1, 2, 3 )
+
+    Triple.Extra.fromList [ 1, 2 ]
+        --> Nothing
+
+-}
+fromList : List a -> Maybe ( a, a, a )
+fromList lst =
+    case lst of
+        a :: b :: c :: _ ->
```

Comment:
From the name alone, I would expect this definition
```elm
case lst of
    [ a, b, c ] -> Just ( a, b, c )
    _ -> Nothing
```
so I'd still prefer a name that makes it clear the elements are from the beginning. All other changes look good!

Reply:
I originally thought of that definition too. But it seemed to me that this definition would be more versatile (although perhaps not - it's a fairly simple fix to do `List.take 3 >> Triple.Extra.fromList`).

Should we then just have the strict version?

Reply:
Well I couldn't tell you a use case for either that wouldn't be nicer as an explicit case of.
But if we decide to keep either, I'm in favor of the `fromListBeginning` taking the first 3

Reply:
OK, let's punt on this one and just remove it for now.